tests/tasks/assert_fail_on_s390x.yml

```
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Assert failure on the s390x architecture
+  assert:
+    that: ansible_architecture == 's390x' in ansible_failed_result.msg
```

```suggestion
    that: "'role does not support s390x' in ansible_failed_result.msg"
```
or - define the string to check for in a vars
```yaml
  assert:
    that: __str in ansible_failed_result.msg
  vars:
    __str: This role does not support s390x architecture
```

---

Right, I am funny, I did only a part of what I actually wanted.
Now I check correctly, I also register a variable to then do the actual clean up only when this assertion task is skipped (the system is not s390x)

---

Actually, it's better to end host right in assert_fail_on_s390x.yml